redis/src/storages/redis/redis_secdist.cpp

```diff
@@ -38,6 +39,14 @@ RedisMapSettings::RedisMapSettings(const formats::json::Value& doc) {
             ? USERVER_NAMESPACE::redis::ConnectionSecurity::kTLS
             : USERVER_NAMESPACE::redis::ConnectionSecurity::kNone;
 
+    settings.database_index =
+        client_settings["database_index"].As<std::optional<size_t>>();
```

Comment:
Will not work in cluster mode, would be great if you add a check in `redis/component.cpp`

Reply:
Could you clarify this a little? 

1. Do you want me to extend method [Redis::Connect](https://github.com/userver-framework/userver/blob/develop/redis/src/storages/redis/component.cpp#L195) with  something like this:
```c++
auto sentinel = redis::Sentinel::CreateSentinel(
        thread_pools_, settings, redis_group.config_name, config_source,
        redis_group.db, redis::KeyShardFactory{redis_group.sharding_strategy},
        cc, testsuite_redis_control);
if (sentinel) {
      if (sentinel->IsInClusterMode() && settings.database_index != 0) {
        // handle it somehow
      }
      ...
}
```
Is it the only one place I need to insert the check or there are some other places as well?
2. How do you want to handle it? Throw an exception or just skip it with the corresponding log message like when sentinel pointer turns out to be null?
3. Are you sure you want to add checks in this case? As you have mentioned, cluster setup does not support selecting databases, so service will shut down with the an message like `SELECT failed: unknown command `SELECT``anyway.